### DISCUSSION TITLE
How does umami detect client disconnection/exit event?

### DISCUSSION THREAD

**Discussion by daniel31x13** [umami-software/umami#950] [Q&A] [answered]
Hello 👋👋
Knowing when the user left the webpage is required to check the realtime active users and amount of activity spent on website.
How are they measured?

**Comment by mikecao** [comment]
There is no accurate way to determine that at the moment. Right now we just take the delta between page visits. But if they visit only one page we can't tell.

**Reply by daniel31x13** [reply]
Oh right so its kinda like whats Google Analytics doing, may I ask why don't you just create a websocket for each visitor?? I suppose wouldn't that be far more accurate?

**Comment by mikecao** [comment, marked as answer]
Because the websocket wouldn't persist when they move between pages anyways. The answer is to use the Beacon API, just haven't got around to it yet.

**Reply by daniel31x13** [reply]
Tho the fact that I got my answer (Thanks a lot.) 
I also had one last question: Is Websocket a good way of doing this? Because I am also doing something similar and I get so many judgement that this procedure is resource hungry at scale.

**Reply by mikecao** [reply]
Yes websockets are resource heavy. Plus you need a server to process the socket connections. For tracking visits you would need to open a new socket connect for every page visit.

**Reply by daniel31x13** [reply]
Oh I see, really appreciate the help! Thank you 🙏
